src/biokbase/narrative/tests/generate_test_results.py

```diff
@@ -0,0 +1,219 @@
+import copy
+from biokbase.narrative.tests.util import ConfigTests
+from biokbase.narrative.app_util import app_version_tags
+
+from biokbase.narrative.jobs.jobcomm import (
+    INFO,
+    LOGS,
+    RETRY,
+    STATUS,
+)
+
+from biokbase.narrative.jobs.job import (
+    JOB_ATTR_DEFAULTS,
+    EXCLUDED_JOB_STATE_FIELDS,
+    COMPLETED_STATUS,
+)
+from biokbase.narrative.tests.job_test_constants import (
+    BAD_JOBS,
+    RETRIED_JOBS,
+    BATCH_PARENT,
+    JOB_COMPLETED,
+    BATCH_COMPLETED,
+    BATCH_RETRY_COMPLETED,
+    get_test_job,
+    generate_error,
+)
+
+config = ConfigTests()
+
+TEST_JOBS = config.load_json_file(config.get("jobs", "ee2_job_test_data_file"))
+spec_list = config.load_json_file(config.get("specs", "app_specs_file"))
+TEST_SPECS = {}
+
+for tag in app_version_tags:
+    spec_dict = dict()
+    for spec in spec_list:
+        spec_dict[spec["info"]["id"]] = spec
+    TEST_SPECS[tag] = spec_dict
+
+
+def get_test_spec(tag, app_id):
+    return copy.deepcopy(TEST_SPECS[tag][app_id])
+
+
+def _generate_job_output(job_id):
+    state = get_test_job(job_id)
+    job_input = state.get("job_input", {})
+    narr_cell_info = job_input.get("narrative_cell_info", {})
+
+    state.update(
+        {
+            "batch_id": state.get(
+                "batch_id", job_id if state.get("batch_job", False) else None
+            ),
+            "cell_id": narr_cell_info.get("cell_id", None),
+            "run_id": narr_cell_info.get("run_id", None),
+            "job_output": state.get("job_output", {}),
+            "child_jobs": state.get("child_jobs", []),
+        }
+    )
+    for f in EXCLUDED_JOB_STATE_FIELDS:
+        if f in state:
+            del state[f]
+
+    if state["status"] != COMPLETED_STATUS:
+        return {"jobState": state, "outputWidgetInfo": None}
+
+    widget_info = {
```

Comment:
Not very nice hardcoded job data -- generating the output widget info uses a number of functions from the Job object, and for now I have just copied the output of those functions rather than examining them and checking if there's an easy way to mock them for tests. Should be easy enough to retrofit mocked data if necessary.

Reply:
I think this is probably fine for now. We've used that format forever, and I don't see it changing anytime soon.